factory/__main__.py

```diff
@@ -1,37 +1,79 @@
-import os
 import signal
+import argparse
 import sys
+import os
 from pathlib import Path
 
 from ImageGoNord import GoNord
 
 from rich.console import Console
 from rich.panel import Panel
-from rich.prompt import IntPrompt
-from rich.progress import track
+
 
 def main():
-    signal.signal(signal.SIGINT, signal_handler)
-    console = Console()
 
-    global gruvbox_factory
+    global gruvbox_factory, console
     gruvbox_factory = GoNord()
     gruvbox_factory.reset_palette()
     add_gruvbox_palette()
 
+    signal.signal(signal.SIGINT, signal_handler)
+    console = Console()
+
+    # Checks if there's an argument
+    if len(sys.argv) > 1:
+        image_file = fromCommandArgument()
+    else:
+        image_file = fromTui()
+    process_image(image_file)
+
+
+
+# Gets the file path from the Argument
+def fromCommandArgument():
+
+    command_parser = argparse.ArgumentParser(
+        description='A simple cli to convert manufacture a gruvbox themed wallpaper;'
+    )
+    command_parser.add_argument('Path',
+                       metavar='path',
+                       type=str,
+                       help='The path to the image.')
+    args = command_parser.parse_args()
+
+    if args.Path is None:
+        console.print('❌ [red]We had a problem in the pipeline! Make sure you\'re in the same path of the image you want to convert! [/]')
+        sys.exit(0)
+    else:
+        return args.Path
+
+
+# Gets the file path from user input
+def fromTui():
+
     console.print(Panel('🏭 [bold green] Gruvbox Factory [/] 🏭', expand=False, border_style='yellow'))
     console.print('⚠️ WARNING ⚠️\n[italic]make sure you\'re in the same directory of the image you want to convert [/]\n')
-    image_file = console.input('🖼️ [bold yellow]Which image do you want to manufacture?[/] ')
+    return console.input('🖼️ [bold yellow]Which image do you want to manufacture?[/] ')
 
+def process_image(image_file):
     try:
         image = gruvbox_factory.open_image(image_file)
     except:
-        console.print('❌ [red]We had a problem in the pipeline! Make sure you\'re in the same path of the image you want to convert! [/]')
+        console.print('❌ [red]We had a problem in the pipeline! Make sure you\'re in the same path of the image you want to convert or the Path is correct! [/]')
         sys.exit(0)
     console.print('🔨 [yellow]manufacturing your gruvbox wallpaper...[/]')
-    gruvbox_factory.convert_image(image, save_path=('gruvbox_' + image_file))
+    
+
+    #TODO: might be a better idea to save the new Image in the same directory the command is being run from
```

Comment:
hmm, I think it might be a good idea too :thinking: Actualy, the perfect way would be having a default directory to save and give the user the option to save it anywhere else